training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -40,6 +43,16 @@
     @Autowired
     PasswordEncoder passwordEncoder;
 
+    @GetMapping("/username")
+    public User currentUserName(Authentication authentication) throws UserNotFoundException {
+        User user = userRepository.findFirstByUsername(authentication.getName());
+
+        if(user==null){
```

Comment:
This validation shouldn't be necessary, if a user is logged in then it must exist in the repository :) 
What you could validate is what happens when the user is not logged in - would "authentication" be null? If that's so, the clause authentication.getName() will throw NullPointerException.

Reply:
Deleted that validation. If the user is not logged in, it throws unauthorized directly (made a test for that :smile:).

Reply:
Awesome! Nice job 😊